modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/IgniteTableImpl.java

```diff
@@ -111,6 +114,9 @@ public class IgniteTableImpl extends AbstractTable implements IgniteTable, Updat
 
     private final PartitionExtractor partitionExtractor;
 
+    /** Triggers statistic update. */
+    private static AtomicBoolean updateStat = new AtomicBoolean();
```

Comment:
It updates in IgniteTableImpl#handleInsertResults

Reply:
yes, I see this. But still... You have _static_ field that initialised to `false` (default value for AtomicBoolean). You change this field to `true` once the table has processed the very first insert from SQL (but not from KV). 

After commit `e9b8fbf` this has even less sense, because you initialised this field to `true`, thus this field literally does nothing.

Reply:
got it, fixed